src/k8s/pkg/k8sd/app/hooks_join.go

```diff
@@ -40,80 +37,40 @@ func onPostJoin(s *state.State, initConfig map[string]string) error {
 		return fmt.Errorf("failed to get IP address(es) from ServiceCIDR %q: %w", cfg.Network.ServiceCIDR, err)
 	}
 
-	switch cfg.APIServer.Datastore {
-	case "k8s-dqlite":
-		certificates := pki.NewK8sDqlitePKI(pki.K8sDqlitePKIOpts{
-			Hostname: s.Name(),
-			IPSANs:   []net.IP{{127, 0, 0, 1}},
-			Years:    20,
-		})
-		certificates.K8sDqliteCert = cfg.Certificates.K8sDqliteCert
-		certificates.K8sDqliteKey = cfg.Certificates.K8sDqliteKey
-		if err := certificates.CompleteCertificates(); err != nil {
-			return fmt.Errorf("failed to initialize cluster certificates: %w", err)
-		}
-		if err := setup.EnsureK8sDqlitePKI(snap, certificates); err != nil {
-			return fmt.Errorf("failed to write cluster certificates: %w", err)
-		}
-	case "external":
-		certificates := &pki.ExternalDatastorePKI{
-			DatastoreCACert:     cfg.Certificates.DatastoreCACert,
-			DatastoreClientCert: cfg.Certificates.DatastoreClientCert,
-			DatastoreClientKey:  cfg.Certificates.DatastoreClientKey,
-		}
-		if err := certificates.CheckCertificates(); err != nil {
-			return fmt.Errorf("failed to initialize cluster certificates: %w", err)
-		}
-		if err := setup.EnsureExtDatastorePKI(snap, certificates); err != nil {
-			return fmt.Errorf("failed to write cluster certificates: %w", err)
-		}
-	default:
-		return fmt.Errorf("unsupported datastore %s, must be one of %v", cfg.APIServer.Datastore, setup.SupportedDatastores)
+	// TODO check if join node gets selfsigned cert allowed
```

Comment:
no, all k8s-dqlite nodes must use the same certificates. see my other comment, perhaps it's best to keep the certificate generation as is for now.

Reply:
Agreed it was getting way to messy, I left them as is now.